client/src/components/BubblePage.js

```diff
@@ -3,11 +3,23 @@ import axios from "axios";
 
 import Bubbles from "./Bubbles";
 import ColorList from "./ColorList";
+import { axiosWithAuth } from "../utils/AxiosWithAuth";
 
 const BubblePage = () => {
   const [colorList, setColorList] = useState([]);
   // fetch your colors data from the server when the component mounts
   // set that data to the colorList state property
+  useEffect(() => {
+    axiosWithAuth()
+      .get(`/api/colors`)
+      .then(res => {
+        setColorList(res.data);
+        console.log(colorList)
+      })
+      .catch(err => {
+        console.log('Error in GET post api', err.response)
+      })
+  }, [])
```

Comment:
Two ways you could write this code:
```
// axios call inside the useEffect()
useEffect(() => {
  axiosWithAuth()
    .get(`/api/colors`)
    .then(res => {
      setColorList(res.data);
      console.log(colorList)
    })
    .catch(err => {
    console.log('Error in GET post api', err.response)
    })
}, [])
```
OR
```

// axios call inside a function
const fetchColors = () => {
  axiosWithAuth()
  .get(`/api/colors`)
  .then(res => {
    setColorList(res.data);
    console.log(colorList)
  })
    .catch(err => {
    console.log('Error in GET post api', err.response)
  })
}

// function called inside useEffect()
useEffect(() => {
  fetchColors()
}, [])
```

If we use the second method, our axios call exists as a function that we can reuse as many times as we want without it being stuck inside our `useEffect()` hook.